pennylane/_device.py

```diff
@@ -302,33 +309,69 @@ def supports_observable(self, observable):
         if isinstance(observable, str):
             return observable in self.observables
 
-        raise ValueError("The given operation must either be a pennylane.Observable class or a string.")
+        raise ValueError("The given observable must either be a pennylane.Observable class or a string.")
+
+    def get_original_operation_name(self, operation_name: str):
+        """Given an inverse operation, returns the name of the original. First, it checks if the
+        device supports inverse operations.
+
+        Args:
+            operation_name (str): name of the inverse operation
+
+        Raises:
+            DeviceError: if the device does not support the computation of inverses
+
+        Returns:
+            original_name (str): name of the original operation
+        """
+        if not self.capabilities().get("inverse_operations", False):
+            raise DeviceError("The inverse of gates are not supported on device {}".format(self.short_name))
+        return operation_name[:-len(Operation.string_for_inverse)]
```

Comment:
I would change this implementation by adding a property `original_name` to the  `Operation` class. It feels weird to do this post_processing with "internals" from the `Operation` class outside of it.

Reply:
Also, the "original name" is manifestly a property of the Operation itself and should thus be implemented there for reasons of encapsulation.

Reply:
How about `base_name`, or just using `op.class.__name__`?

Reply:
`base_name = op.class.__name__` is good!